### PR TITLE
Add pip check to test.yml

### PR DESCRIPTION
**Pull request by yiboliu**
### Reasons for making this change
Check if there's any broken version requirements

### Related issues
#3058 

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by nelson-liu** [comment]
hm, this should be failing, but it isn't, because i think it's running outside of the virtualenv that is set up by `pre-commit.sh`. Maybe we should add pip check to `pre-commit.sh` instead?

Also, might be good to do the `pip install`s in `pre-commit.sh` with `--no-cache`

File: pre-commit.sh [hunk 1]
```diff
@@ -9,9 +9,15 @@ if ! [ -e venv ]; then
...
   python3 -m virtualenv -p python3 venv
 fi
-venv/bin/pip install -r requirements-server.txt
-venv/bin/pip install -r requirements.docs.txt
-venv/bin/pip install -r requirements.dev.txt
+venv/bin/pip install -r requirements-server.txt --no-cache
+venv/bin/pip install -r requirements.docs.txt --no-cache
+venv/bin/pip install -r requirements.dev.txt --no-cache
+
+venv/bin/pip check || let val=$?
```
Comment (nelson-liu) [inline pre-commit.sh]: you don't need the `||` or anything below since the script runs with `set -e`. When `pip check` exits with 1, the whole script will.

**Comment by nelson-liu** [comment]
Looks like this still fails because there's a conflict in the package version for mock, which we pin and apache beam also pins.

File: pre-commit.sh [hunk 2]
```diff
@@ -9,9 +9,11 @@ if ! [ -e venv ]; then
...
   python3 -m virtualenv -p python3 venv
 fi
-venv/bin/pip install -r requirements-server.txt
-venv/bin/pip install -r requirements.docs.txt
-venv/bin/pip install -r requirements.dev.txt
+venv/bin/pip install -r requirements-server.txt --no-cache
+venv/bin/pip install -r requirements.docs.txt --no-cache
+venv/bin/pip install -r requirements.dev.txt --no-cache
+
+venv/bin/pip check
```
Comment (epicfaace) [inline pre-commit.sh]: Should we run `pip check` after packages are installed with `pip install -e .`?

File: pre-commit.sh [hunk 3]
```diff
@@ -9,12 +9,14 @@ if ! [ -e venv ]; then
   python3 -m pip install virtualenv
   python3 -m virtualenv -p python3 venv
 fi
-venv/bin/pip install -r requirements-server.txt
-venv/bin/pip install -r requirements.docs.txt
-venv/bin/pip install -r requirements.dev.txt
+venv/bin/pip install -r requirements-server.txt --no-cache
```
Comment (epicfaace) [inline pre-commit.sh]: why did you add --no-cache? isn’t it faster / easier if we don’t have that argument in?

File: requirements-server.txt [hunk 4]
```diff
@@ -11,7 +11,7 @@ boto3==1.15.7
 azure-batch==9.0.0
 
 # For testing
-mock==3.0.5
+mock==2.0.0
```
Comment (epicfaace) [inline requirements-server.txt:14]: Why do we need to downgrade mock? Was this causing dependency conflict issues?

Comment (yiboliu) [reply requirements-server.txt:14]:
```
apache-beam 2.25.0 requires mock<3.0.0,>=1.0.1, but you'll have mock 3.0.5 which is incompatible.
```

File: pre-commit.sh (hunk 3 above)
Comment (nelson-liu) [reply pre-commit.sh]: I think that without `--no-cache`, you might inadvertently install an older version of the package if it happens to be on your system. Might also need `-U` on these installs, so the venv automatically gets updated if the pins are relaxed..

Comment (yiboliu) [reply pre-commit.sh]: We have version specifications right? Each module is specified to a particular version.

**Review by epicfaace — APPROVED** [review]
